code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -287,31 +291,31 @@
 	air_contents.gases[/datum/gas/nitrogen][MOLES] = (N2STANDARD * maximum_pressure * filled) * air_contents.volume / (R_IDEAL_GAS_EQUATION * air_contents.temperature)
 
 /obj/machinery/portable_atmospherics/canister/update_icon_state()
-	if(machine_stat & BROKEN)
+	if(machine_stat & BROKEN && !has_been_broken)
 		icon_state = "[icon_state]-1"
-
+		has_been_broken = TRUE
+	else
+		has_been_broken = FALSE
```

Comment:
Instead of a `has_been_broken = FALSE` var I'd define something like `var/base_icon_state = "tier 1"`
and then do something like
```cpp
/obj/machinery/portable_atmospherics/canister/update_icon_state()
	icon_state = "[base_icon_state]-[machine_stat & BROKEN]"
```
That way you can cleanly append more variations in the future if needed.

Reply:
Yeah this is more what i had in mind with my previous comment. Make it describe the state instead of transfomations.

Reply:
I see, I think I've done what you asked, hope this works.